Artifacts/windows-certificate-from-keyvault/InstallCertificate.ps1

```diff
@@ -0,0 +1,110 @@
+Param(
+    [ValidateNotNullOrEmpty()]
+    [string]$vaultName,
+    [ValidateNotNullOrEmpty()]
+    [string]$secretName,
+    [ValidateNotNullOrEmpty()]
+    [string]$azureServicePrincipalClientId,
+    [ValidateNotNullOrEmpty()]
+    [securestring]$azureServicePrincipalKey,
+    [ValidateNotNullOrEmpty()]
+    [string]$azureServicePrincipalTenantId
+)
+
+# Handle all errors in this script.
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this script, unless you want to ignore a specific error.
+    Handle-LastError
+}
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+$ErrorActionPreference = "Stop"
+
+function Handle-LastError
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $message = $error[0].Exception.Message
+    if ($message)
+    {
+        Write-Host -Object "ERROR: $message" -ForegroundColor Red
+    }
+    
+    # IMPORTANT NOTE: Throwing a terminating error (using $ErrorActionPreference = "Stop") still
+    # returns exit code zero from the PowerShell script when using -File. The workaround is to
+    # NOT use -File when calling this script and leverage the try-catch-finally block and return
+    # a non-zero exit code from the catch block.
+    exit -1
+}
+
+
+try{
+
+
+
+
+    #$securePassword = ConvertTo-SecureString $azurePassword -AsPlainText -Force
+    #$creds = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $azureUsername, $securePassword
+
+    if (-not (Get-Module -Name "AzureRm")){
+        if (Get-Module -ListAvailable | Where-Object { $_.Name -eq "AzureRm"}){
+            
+        }else{
+            Write-Host "AzureRM not detected, installing..."
+            Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+            Install-Module AzureRm -Force -AllowClobber
+        }
+    }
+
+    Import-Module AzureRm
+
+    $azureAccountName = $azureServicePrincipalClientId
+    #$azurePassword = ConvertTo-SecureString $azureServicePrincipalKey -AsPlainText -Force
+    $psCred = New-Object System.Management.Automation.PSCredential($azureAccountName, $azureServicePrincipalKey)
+
+    Write-Host "Logging into Azure"
+    Add-AzureRmAccount -Credential $psCred -TenantId $azureServicePrincipalTenantId -ServicePrincipal
+    #Add-AzureRmAccount -Credential $creds
```

Comment:
Fixed in https://github.com/Azure/azure-devtestlab/pull/248/commits/42f85103f9c4e29f4e1c16c767eeeed8689bcdf2